src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -865,22 +884,7 @@ protected void runMayThrow() throws Exception
 
     public void forceUserDefinedCompaction(String dataFiles)
     {
-        String[] filenames = dataFiles.split(",");
-        Multimap<ColumnFamilyStore, Descriptor> descriptors = ArrayListMultimap.create();
-
-        for (String filename : filenames)
-        {
-            // extract keyspace and columnfamily name from filename
-            Descriptor desc = Descriptor.fromFilename(filename.trim());
-            if (Schema.instance.getCFMetaData(desc) == null)
-            {
-                logger.warn("Schema does not exist for file {}. Skipping.", filename);
-                continue;
-            }
-            // group by keyspace/columnfamily
-            ColumnFamilyStore cfs = Keyspace.open(desc.ksname).getColumnFamilyStore(desc.cfname);
-            descriptors.put(cfs, cfs.getDirectories().find(new File(filename.trim()).getName()));
-        }
+        Multimap<ColumnFamilyStore, Descriptor> descriptors =Descriptor.fromFilenamesGrouped(Arrays.asList(dataFiles.split(",")));
```

Comment:
this logic was repeated in a few places, so I moved it into a helper:  `Descriptor.fromFilenamesGrouped`